src/MagnumPlugins/AssimpImporter/AssimpImporter.cpp

```diff
@@ -1118,6 +1135,235 @@ Containers::Optional<ImageData2D> AssimpImporter::doImage2D(const UnsignedInt id
     return importer->image2D(0, level);
 }
 
+UnsignedInt AssimpImporter::doAnimationCount() const {
+    /* If the animations are merged, there's at most one */
+    if(configuration().value<bool>("mergeAnimationClips"))
+        return _f->scene->mNumAnimations ? 1 : 0;
+
+    return _f->scene->mNumAnimations;
+}
+
+Int AssimpImporter::doAnimationForName(const std::string& name) {
+    /* If the animations are merged, don't report any names */
+    if(configuration().value<bool>("mergeAnimationClips")) return -1;
+
+    if(!_f->animationIndicesForName) {
+        _f->animationIndicesForName.emplace();
+        _f->animationIndicesForName->reserve(_f->scene->mNumAnimations);
+        for(std::size_t i = 0; i != _f->scene->mNumAnimations; ++i)
+            _f->animationIndicesForName->emplace(std::string(_f->scene->mAnimations[i]->mName.C_Str()), i);
+    }
+
+    const auto found = _f->animationIndicesForName->find(name);
+    return found == _f->animationIndicesForName->end() ? -1 : found->second;
+}
+
+std::string AssimpImporter::doAnimationName(UnsignedInt id) {
+    /* If the animations are merged, don't report any names */
+    if(configuration().value<bool>("mergeAnimationClips")) return {};
+    return _f->scene->mAnimations[id]->mName.C_Str();
+}
+
+namespace {
+
+Animation::Extrapolation extrapolationFor(aiAnimBehaviour behaviour) {
+    switch (behaviour) {
+        case aiAnimBehaviour_DEFAULT:
+            /** @todo emulate this correctly by inserting keyframes
+                with the default node transformation */
```

Comment:
Hmm, or the Animation lib could support this directly? 

IIRC I was considering this option but it felt kinda useless to me (couldn't think of a practical use case, since "default transformation" is kinda arbitrary and can mean anything) so I didn't add it.

Reply:
I suppose you could add the concept of "border" values for extrapolation. For this particular thing it seems a bit useless, since the default node translation/rotation/etc. is well-defined and the model *should* have a key-frame at 0.

Reply:
Border values sound useful, but from a technical perspective the more flexible the extrapolation gets, the slower the animation processing will be due to all the special-casing... so if it's doable, I think adding extra keyframes at the beginning/end is the better solution :)

Reply:
Would you be OK with leaving the todo in there for now and postponing this to a later fix? I haven't found a real-world file that even relies on this behaviour, and it complicates the import code and tests quite a bit.

Reply:
Absolutely :) A TODO is fine.